src/Equinor.ProCoSys.IPO.WebApi/Authentication/Authenticator.cs

```diff
@@ -40,14 +41,38 @@ public async ValueTask<string> GetBearerTokenOnBehalfOfCurrentUserAsync()
                         .AcquireTokenOnBehalfOf(new List<string> { _options.Value.MainApiScope }, new UserAssertion(_requestToken.ToString()))
                         .ExecuteAsync();
 
-                    _onBehalfOfUserToken = tokenResult.AccessToken;
+                    _onBehalfOfUserTokenForMainApi = tokenResult.AccessToken;
                 }
-                return _onBehalfOfUserToken;
+                return _onBehalfOfUserTokenForMainApi;
             }
-            else
+
+            return _requestToken;
+        }
+
+        public async ValueTask<string> GetBearerTokenForLibraryApiOnBehalfOfCurrentUserAsync()
+        {
+            if (_canUseOnBehalfOf)
             {
-                return _requestToken;
+                if (_onBehalfOfUserTokenForLibraryApi == null)
+                {
```

Comment:
Extract as a method injecting scope, as that is the only difference for mainapi and libraryapi scope